plugins/field-colour/src/blocks/colourRandom.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Block, common as BlocklyCommon} from 'blockly';
+import {
+  JavascriptGenerator,
+  Order as JavascriptOrder,
+} from 'blockly/javascript';
+import {DartGenerator, Order as DartOrder} from 'blockly/dart';
+import {LuaGenerator, Order as LuaOrder} from 'blockly/lua';
+import {PhpGenerator, Order as PhpOrder} from 'blockly/php';
+import {PythonGenerator, Order as PythonOrder} from 'blockly/python';
+import {registerFieldColour} from '../field_colour';
+import {Generators} from './generatorUtils';
+
+/** The name this block is registered under. */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const BLOCK_NAME = 'colour_random';
+
+// Block for random colour.
+const jsonDef = {
+  type: BLOCK_NAME,
+  message0: '%{BKY_COLOUR_RANDOM_TITLE}',
+  output: 'Colour',
+  helpUrl: '%{BKY_COLOUR_RANDOM_HELPURL}',
+  style: 'colour_blocks',
+  tooltip: '%{BKY_COLOUR_RANDOM_TOOLTIP}',
+};
+
+/**
+ * Javascript block generator function.
+ *
+ * @param block The Block instance to generate code for.
+ * @param generator The JavascriptGenerator calling the function.
+ * @returns A tuple containing the code string and precedence.
+ */
+export function jsGenerator(
+  block: Block,
+  generator: JavascriptGenerator,
+): [string, JavascriptOrder] {
+  // Generate a random colour.
+  const functionName = generator.provideFunction_(
+    'colourRandom',
+    `
+function ${generator.FUNCTION_NAME_PLACEHOLDER_}() {
+  var num = Math.floor(Math.random() * Math.pow(2, 24));
```

Comment:
I think it would be preferable to use a constant rather than calling `Math.pow` every time this block is invoked:
```suggestion
  var num = Math.floor(Math.random() * 0x1000000);
```

Reply:
Tracked in #2198 for follow-up.